### PR TITLE
[plugin.video.cbc] 4.0.6

### PR DESCRIPTION
**Pull request by micahg**
* switch to inputstream.adaptive for hls
* add fallback to ffmpeg option

### Checklist:

- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-plugins/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

### CODE REVIEW

File: plugin.video.cbc/addon.xml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<addon id="plugin.video.cbc"
+       name="Canadian Broadcasting Corp (CBC)"
+       version="4.0.5"
+       provider-name="micahg,t1m">
+  <requires>
+    <import addon="xbmc.python" version="2.20.0"/>
```
Comment (Rechi) [inline plugin.video.cbc/addon.xml]: > WARN: For leia it is advised to set xbmc.python version to 2.26.0

File: plugin.video.cbc/resources/lib/cbc.py [hunk 2]
```diff
@@ -0,0 +1,216 @@
+import requests, uuid, urllib, json
+from xml.dom.minidom import *
+import xml.etree.ElementTree as ET
+
+from utils import saveCookies, loadCookies, saveAuthorization, log
```
Comment (Rechi) [inline plugin.video.cbc/resources/lib/cbc.py]:
```suggestion
from .utils import saveCookies, loadCookies, saveAuthorization, log
```

File: plugin.video.cbc/resources/lib/livechannels.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import requests, json
+from utils import saveCookies, loadCookies, log
```
Comment (Rechi) [inline plugin.video.cbc/resources/lib/livechannels.py]:
```suggestion
from .utils import saveCookies, loadCookies, log
```

File: plugin.video.cbc/resources/lib/liveprograms.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
+import requests, json
+from utils import saveCookies, loadCookies, log
```
Comment (Rechi) [inline plugin.video.cbc/resources/lib/liveprograms.py]:
```suggestion
from .utils import saveCookies, loadCookies, log
```

File: plugin.video.cbc/resources/lib/shows.py [hunk 5]
```diff
@@ -0,0 +1,161 @@
+import requests, urllib
+from xml.dom.minidom import *
+from utils import saveCookies, loadCookies, loadAuthorization, log
```
Comment (Rechi) [inline plugin.video.cbc/resources/lib/shows.py]:
```suggestion
from .utils import saveCookies, loadCookies, loadAuthorization, log
```

File: plugin.video.cbc/resources/lib/utils.py [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+    Log an error
+    @param msg The error to log
+    @param error error severity indicator
+    """
+    try:
+        import xbmc
+        full_msg = "plugin.video.cbc: {}".format(msg)
+        xbmc.log(full_msg, level=xbmc.LOGERROR if error else xbmc.LOGINFO)
+    except:
+        print msg
```
Comment (Rechi) [inline plugin.video.cbc/resources/lib/utils.py]:
```suggestion
        print(msg)
```

File: plugin.video.cbc/default.py [hunk 7]
```diff
@@ -0,0 +1,246 @@
...
+    prog.close()
+    return True
+
+
+def play(labels, image, url):
+    item = xbmcgui.ListItem(labels['title'], path=url)
+    item.setArt({ 'thumb': image, 'poster': image })
+    item.setInfo(type="Video", infoLabels=labels)
+    helper = inputstreamhelper.Helper('hls')
+    if xbmcaddon.Addon().getSetting("ffmpeg") == 'false' and helper.check_inputstream():
```
Comment (Rechi) [inline plugin.video.cbc/default.py]:
```suggestion
    if not xbmcaddon.Addon().getSettingBool("ffmpeg") and helper.check_inputstream():
```

**Comment by micahg** [comment]
@Rechi I'll mark the old one as broken -- does it need a version bump when I do so? Can it also have 4.0.5 on that branch?

Thanks

**Comment by Rechi** [comment]
Yes every change, also marking an add-on broken, needs a version bump.
While it would work to use 4.0.5 for the broken version and the updated leia version, I recommand to update this at least to 4.0.6. Then there is no guessing which 4.0.5 variant one actually has.

**Comment by micahg** [comment]
OK, thanks @Rechi -- requested changes made.

**Comment by micahg** [comment]
tyvm.

**Pull request merged**